### PR TITLE
Batch processing of beam search

### PR DESCRIPTION
**Pull request by mn5k**
| Batchsize | ATT | CTC | RNNLM | WER (test_clean) |
----|----|----|----|---- 
| 1 (orig.)  |  ✓   |   ✓  |  ✓  |  6.5    |
| 48  | ✓   |   ✓  |  ✓  |  6.5    |

### CODE REVIEW

File: src/bin/asr_train.py [hunk 1]
```diff
@@ -105,6 +105,25 @@ def main():
                         help='Apply label smoothing with a specified distribution type')
     parser.add_argument('--lsm-weight', default=0.0, type=float,
                         help='Label smoothing weight')
+    # recognition related
```
Comment (takaaki-hori) [inline src/bin/asr_train.py]: A bit confusing. It's better to say `recognition options to compute CER/WER`

**Comment by sw005320** [comment]
Can you please revisit it? This function is really important.

**Comment by mn5k** [comment]
Thank you for the review.
I'll modify the code and add an option for word-based LM.

**Comment by mn5k** [comment]
@takaaki-hori There is a class (MultiLevelLM, #235) that combines character-based and word-based language models in extlm_pytorch.py. Could you tell me the WER of this method?
The WER of test_eval92 was high as 102.6% in my condition.

**Comment by takaaki-hori** [comment]
Maybe something wrong.
In my test, I got a bit better result than LookAheadLM in WSJ with a 20k-word LM.
But I have not tested it recently. I will revisit it in #406 .

**Comment by mn5k** [comment]
Thank you for the confirmation
.   solved the problem.

**Comment by mn5k** [comment]
For the computation of word-RNNLM and char/word-RNNLM, I added a buffer for sample-level processing and concatenation of samples. I'll consider a smarter way later.

**Comment by mn5k** [comment]
I used editdistance in
https://github.com/aflc/editdistance
for the computation of CER/WER now.
Should I implement it without the external package?

**Comment by sw005320** [comment]
I want to restart this project. It seems that we don't have serious issue, and we may merge it. @mn5k, do you have any concerns?

**Comment by mn5k** [comment]
I think there's no problem.

**Comment by sw005320** [comment]
we can use batch decoding as default. maybe 1 is enough. then could you add the test to check batch and non batch decodings yield the same results?

**Comment by sw005320** [comment]
Thanks a lot! I merged it.

**Comment by mn5k** [comment]
I added a test code to check the output of two search algorithms.

As I mentioned before, my search algorithm does not fully correspond to the original search.
I only checked the search without a fusion of other external modules.

**Pull request merged**